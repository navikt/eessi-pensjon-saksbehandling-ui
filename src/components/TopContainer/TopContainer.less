@import '../../constants';

.c-topContainer {
    background-color: @white;
    display: flex;
    flex-direction: column;
    min-height: 100vh;

    ._container {
        flex: 1 0 auto;
        display: flex;
        flex-direction: column;
    }

    &.highContrast {

        color: @high-contrast-orange !important;
        background-color: @black !important;
    }
}
