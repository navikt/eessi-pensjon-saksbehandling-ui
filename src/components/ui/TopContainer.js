import React, { Component } from 'react';
import PT from 'prop-types';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { bindActionCreators }  from 'redux';

import * as Nav from './Nav';
import TopHeader from './Header/TopHeader';
import ServerAlert from './Alert/ServerAlert';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import Modal from './Modal';

import * as statusActions from '../../actions/status';

const mapStateToProps = () => {
    return {}
};

const mapDispatchToProps = (dispatch) => {
    return {actions: bindActionCreators(Object.assign({}, statusActions), dispatch)};
};

class TopContainer extends Component {

    componentDidMount() {

        const { actions, location } = this.props;

        const rinaId = new URLSearchParams(location.search).get('rinaId');

        actions.setRinaId(rinaId); // if rinaId is undefined, it erases rinaId

        if (rinaId) {
            actions.getStatus(rinaId);
            actions.getCase(rinaId);
        }
    }

    render () {

        const { className, style, history } = this.props;

        return <div style={style} className={classNames('topcontainer', className)}>
            <TopHeader/>
            <ServerAlert/>
            <Modal/>
            <Breadcrumbs history={history}/>
            <Nav.Container fluid={true}>
                {this.props.children}
            </Nav.Container>
        </div>
    }
}

TopContainer.propTypes = {
    children  : PT.node.isRequired,
    className : PT.string,
    style     : PT.object,
    actions   : PT.object.isRequired,
    history   : PT.object.isRequired,
    location  : PT.object.isRequired
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(TopContainer)

