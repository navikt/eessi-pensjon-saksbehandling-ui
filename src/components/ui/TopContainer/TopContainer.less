@import '../../../constants';

.c-ui-topContainer {
    height: 100%;

    ._container {
        flex: 1 0 auto;
    }

    &.SAKSBEHANDLER {
        background-color: @white-off;
    }

    &.BRUKER {
        background-color: @white;
    }
}
