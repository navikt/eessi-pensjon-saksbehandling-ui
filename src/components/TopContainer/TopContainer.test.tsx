import { alertClear } from 'actions/alert'
import { closeModal } from 'actions/ui'
import { ModalContent } from 'declarations/components'
import { mount, ReactWrapper } from 'enzyme'
import { TopContainer, TopContainerProps, TopContainerSelector } from './TopContainer'
import { stageSelector } from 'setupTests'

/* jest.mock('use-error-boundary', () => ({
  __esModule: true, // this property makes it work
  default: () => ({
    ErrorBoundary: ({ children }: any) => <div className='mock-error-boundary'>{children}</div>
  })
})) */

const mockHistoryPush = jest.fn()

jest.mock('react-router-dom', () => {
  return {
    useHistory: () => ({
      push: mockHistoryPush
    })
  }
})

const defaultSelector: TopContainerSelector = {
  clientErrorParam: undefined,
  clientErrorStatus: 'ERROR',
  clientErrorMessage: 'mockErrorMessage',
  serverErrorMessage: undefined,
  error: undefined,
  expirationTime: undefined,
  footerOpen: false,
  gettingUserInfo: false,
  highContrast: false,
  isLoggingOut: false,
  modal: undefined,
  params: {},
  size: 'lg',
  username: 'mockUsername'
}

jest.mock('actions/alert', () => ({
  alertClear: jest.fn(),
  alertFailure: jest.fn()
}))

jest.mock('actions/ui', () => ({
  closeModal: jest.fn(),
  toggleHighContrast: jest.fn()
}))

describe('components/TopContainer', () => {
  let wrapper: ReactWrapper
  const initialMockProps: TopContainerProps = {
    header: 'mockHeader'
  }

  beforeEach(() => {
    stageSelector(defaultSelector, {})
    wrapper = mount(
      <TopContainer {...initialMockProps}>
        <div id='TEST_CHILD' />
      </TopContainer>
    )
  })

  it('Render: match snapshot', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    expect(wrapper.exists('Header')).toBeTruthy()
    expect(wrapper.exists('Alert')).toBeTruthy()
    expect(wrapper.exists('SessionMonitor')).toBeTruthy()
    expect(wrapper.exists('Footer')).toBeTruthy()
    expect(wrapper.exists('Modal')).toBeFalsy()
  })

  it('Render: client error message', () => {
    (alertClear as jest.Mock).mockReset()
    stageSelector(defaultSelector, { clientErrorMessage: 'mockMessage|mockParams' })
    wrapper = mount(
      <TopContainer {...initialMockProps}>
        <div id='TEST_CHILD' />
      </TopContainer>
    )

    expect(wrapper.find('.alertstripe__tekst').hostNodes().render().text()).toEqual('mockMessage|mockParams')
    wrapper.find('[data-test-id=\'c-alert__close-icon\']').hostNodes().simulate('click')
    expect(alertClear).toHaveBeenCalled()
  })

  it('Handling: open / closes modal', () => {
    expect(wrapper.exists('Modal')).toBeFalsy()
    const mockModal: ModalContent = {
      modalTitle: 'mockTitle',
      modalText: 'mockText',
      modalButtons: [{
        text: 'ok'
      }]
    }
    stageSelector(defaultSelector, { modal: mockModal })
    wrapper = mount(
      <TopContainer {...initialMockProps}>
        <div id='TEST_CHILD' />
      </TopContainer>
    )

    expect(wrapper.exists('Modal')).toBeTruthy()
    const modal = wrapper.find('Modal').first()
    modal.find('button').hostNodes().last().simulate('click')
    expect(closeModal).toHaveBeenCalled()
  })
})
