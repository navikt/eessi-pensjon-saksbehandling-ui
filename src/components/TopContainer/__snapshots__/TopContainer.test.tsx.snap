// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TopContainer Render: match snapshot 1`] = `
.c17 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g31[id="c16"] {
  content: "c17,";
}

.c23 {
  opacity: 0;
  -webkit-animation: kGLwuF 1s forwards;
  animation: kGLwuF 1s forwards;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 10;
  width: 100%;
  border-radius: 0px !important;
  border: 0px !important;
}

.c23 .alertstripe__tekst {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: none !important;
}

ui-fss.g49[id="c22"] {
  content: "c23,";
}

.c25 {
  position: absolute;
  top: 0.25rem;
  right: 0.25rem;
  cursor: pointer;
}

ui-fss.g50[id="c24"] {
  content: "c25,";
}

.c29 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: white;
  padding: 0rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: width 0.2s ease-out;
  transition: width 0.2s ease-out;
}

.c29.toggled {
  height: 40px;
}

.c29:not(.toggled) {
  height: 10px;
}

ui-fss.g51[id="c28"] {
  content: "c29,";
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c31.fullWidth {
  width: 100%;
}

.c31 .footerButtonClosed {
  width: 100%;
  margin: 0px;
  display: block;
  padding: 0px;
  background-color: white;
}

.c31 .footerButtonClosed:hover {
  cursor: pointer;
  background-color: #FFFFFF;
}

.c31 .footerButtonOpen {
  width: 1.2rem;
  margin-right: 0.5rem;
  display: inline-block;
  padding-top: 0.6rem;
  color: #262626;
  text-align: center;
}

.c31 .footerButtonOpen:hover {
  cursor: pointer;
  background-color: #FFFFFF;
}

ui-fss.g52[id="c30"] {
  content: "c31,";
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 * {
  font-size: inherit;
  line-height: inherit;
}

ui-fss.g60[id="c4"] {
  content: "c5,";
}

.c3 {
  background-color: #262626;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 4rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

ui-fss.g61[id="c2"] {
  content: "c3,";
}

.c11 {
  font-size: 1.5rem;
  color: #FFFFFF;
}

ui-fss.g62[id="c10"] {
  content: "c11,";
}

.c19 {
  margin: auto 0px;
  padding: 0.3em;
}

ui-fss.g63[id="c18"] {
  content: "c19,";
}

.c15 {
  color: white;
}

ui-fss.g64[id="c14"] {
  content: "c15,";
}

.c7 {
  border-left: 1px solid #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 30px;
  width: 1px;
  margin-left: 1rem;
}

ui-fss.g65[id="c6"] {
  content: "c7,";
}

.c9 {
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 13pt;
  padding-left: 15px;
}

ui-fss.g66[id="c8"] {
  content: "c9,";
}

.c13 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

ui-fss.g67[id="c12"] {
  content: "c13,";
}

.c21 {
  padding: 0.45rem;
  padding-left: 0.5rem;
  padding-right: 1.6rem;
  color: #FFFFFF;
}

ui-fss.g68[id="c20"] {
  content: "c21,";
}

.c27 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  background-color: #f1f1f1;
  color: #262626;
}

ui-fss.g83[id="c26"] {
  content: "c27,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

ui-fss.g84[id="c0"] {
  content: "c1,";
}

<TopContainer
  header="mockHeader"
>
  <he
    highContrast={false}
  >
    <Component
      theme={
        Object {
          "alternative-background-color": "#FFFFFF",
          "alternative-hover-color": "#cce1ff",
          "alternative-interactive-color": "#0074df",
          "black": "#000000",
          "border-radius-base": "4px",
          "border-radius-large": "6px",
          "border-radius-small": "2px",
          "container-desktop": "(940px + @grid-gutter-width)",
          "container-large-desktop": "(1140px + @grid-gutter-width)",
          "container-lg": "(1140px + @grid-gutter-width)",
          "container-md": "(940px + @grid-gutter-width)",
          "container-sm": "(720px + @grid-gutter-width)",
          "container-tablet": "(720px + @grid-gutter-width)",
          "fokusFarge": "#00347d",
          "grayBackground": "#f1f1f1",
          "grayIcon": "#a0a0a0",
          "grayInactive": "#6a6a6a",
          "grayModia": "#333333",
          "grid-columns": "12",
          "grid-float-breakpoint": "768px",
          "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
          "grid-gutter-width": "16px",
          "inverted-font-color": "#FFFFFF",
          "line-height-large": "1.3333333",
          "line-height-small": "1.5",
          "main-active-color": "#005b82",
          "main-background-color": "#f1f1f1",
          "main-border-color": "#6a6a6a",
          "main-border-radius": "4px",
          "main-border-width": "1px",
          "main-disabled-color": "#a0a0a0",
          "main-error-color": "#ba3a26",
          "main-focus-color": "#00347d",
          "main-font-color": "#262626",
          "main-hover-color": "#99c3ff",
          "main-interactive-color": "#0067c5",
          "main-success-color": "#06893a",
          "main-warning-color": "#ff9100",
          "navBakgrunn": "#ffffff",
          "navBla": "#0067c5",
          "navBlaDarken20": "#0056b4",
          "navBlaDarken40": "#00459c",
          "navBlaDarken60": "#00347d",
          "navBlaDarken80": "#002252",
          "navBlaLighten20": "#3386e0",
          "navBlaLighten40": "#66a5f4",
          "navBlaLighten60": "#99c3ff",
          "navBlaLighten80": "#cce1ff",
          "navDypBla": "#005b82",
          "navDypBlaDarken20": "#005077",
          "navDypBlaDarken40": "#004367",
          "navDypBlaDarken60": "#003453",
          "navDypBlaDarken80": "#00243a",
          "navDypBlaLighten20": "#3380a5",
          "navDypBlaLighten40": "#66a3c4",
          "navDypBlaLighten60": "#99c4dd",
          "navDypBlaLighten80": "#cce2f0",
          "navGra20": "#c9c9c9",
          "navGra40": "#a0a0a0",
          "navGra60": "#6a6a6a",
          "navGra80": "#4f4f4f",
          "navGraBakgrunn": "#f1f1f1",
          "navGronn": "#06893a",
          "navGronnDarken20": "#007c2e",
          "navGronnDarken40": "#006a23",
          "navGronnDarken60": "#005519",
          "navGronnDarken80": "#003b0f",
          "navGronnLighten20": "#33aa5f",
          "navGronnLighten40": "#66c786",
          "navGronnLighten60": "#99dead",
          "navGronnLighten80": "#ccf1d6",
          "navLilla": "#634689",
          "navLillaDarken20": "#523874",
          "navLillaDarken40": "#412b5d",
          "navLillaDarken60": "#301f46",
          "navLillaDarken80": "#1f142f",
          "navLillaLighten20": "#8269a2",
          "navLillaLighten40": "#a18dbb",
          "navLillaLighten60": "#c0b2d2",
          "navLillaLighten80": "#e0d8e9",
          "navLimeGronn": "#a2ad00",
          "navLimeGronnDarken20": "#939e00",
          "navLimeGronnDarken40": "#7f8900",
          "navLimeGronnDarken60": "#666e00",
          "navLimeGronnDarken80": "#474e00",
          "navLimeGronnLighten20": "#c1cb33",
          "navLimeGronnLighten40": "#d9e366",
          "navLimeGronnLighten60": "#ecf399",
          "navLimeGronnLighten80": "#f9fccc",
          "navLysBla": "#66cbec",
          "navLysBlaDarken20": "#4cadcd",
          "navLysBlaDarken40": "#368da8",
          "navLysBlaDarken60": "#236b7d",
          "navLysBlaDarken80": "#134852",
          "navLysBlaLighten20": "#7cdaf8",
          "navLysBlaLighten40": "#97e6ff",
          "navLysBlaLighten60": "#b5f1ff",
          "navLysBlaLighten80": "#d8f9ff",
          "navLysGra": "#f1f1f1",
          "navMorkGra": "#262626",
          "navOransje": "#ff9100",
          "navOransjeDarken20": "#d47b00",
          "navOransjeDarken40": "#a86400",
          "navOransjeDarken60": "#7d4c00",
          "navOransjeDarken80": "#523300",
          "navOransjeLighten20": "#ffaa33",
          "navOransjeLighten40": "#ffc166",
          "navOransjeLighten60": "#ffd799",
          "navOransjeLighten80": "#ffeccc",
          "navRod": "#c30000",
          "padding-base-horizontal": "12px",
          "padding-base-vertical": "6px",
          "padding-large-horizontal": "16px",
          "padding-large-vertical": "10px",
          "padding-small-horizontal": "10px",
          "padding-small-vertical": "5px",
          "padding-xs-horizontal": "5px",
          "padding-xs-vertical": "1px",
          "pinkErrorBg": "#f3e3e3",
          "redError": "#ba3a26",
          "redErrorDarken20": "#a32a17",
          "redErrorDarken40": "#881d0c",
          "redErrorDarken60": "#6a1204",
          "redErrorDarken80": "#480900",
          "redErrorLighten20": "#d05c4a",
          "redErrorLighten40": "#e18071",
          "redErrorLighten60": "#efa89d",
          "redErrorLighten80": "#f9d2cc",
          "screen-desktop": "992px",
          "screen-lg": "1200px",
          "screen-lg-desktop": "1200px",
          "screen-lg-min": "1200px",
          "screen-md": "992px",
          "screen-md-max": "(@screen-lg-min - 1)",
          "screen-md-min": "992px",
          "screen-sm": "768px",
          "screen-sm-max": "(@screen-md-min - 1)",
          "screen-sm-min": "768px",
          "screen-tablet": "768px",
          "screen-xs-max": "(@screen-sm-min - 1)",
          "type": "theme",
          "white": "#FFFFFF",
        }
      }
    >
      <styled.div
        role="application"
      >
        <div
          className="c0 c1"
          role="application"
        >
          <ErrorBoundary
            FallbackComponent={[Function]}
            onReset={[Function]}
          >
            <ResizeDetector
              handleWidth={true}
              onResize={[Function]}
            >
              <Header
                gettingUserInfo={false}
                isLoggingOut={false}
                targetRef={
                  Object {
                    "current": null,
                  }
                }
                username="mockUsername"
              >
                <styled.header
                  role="banner"
                >
                  <header
                    className="c2 c3"
                    role="banner"
                  >
                    <styled.div>
                      <div
                        className="c4 c5"
                      >
                        <a
                          data-testid="c-header__logo-link"
                          href="#index"
                          onClick={[Function]}
                        >
                          <NavLogoTransparent
                            color="white"
                            height="45"
                            width="100"
                          >
                            <svg
                              height="45"
                              version="1.1"
                              viewBox="0 0 269 169"
                              width="100"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <title>
                                NAV logo
                              </title>
                              <desc>
                                Created with Sketch.
                              </desc>
                              <defs>
                                <polygon
                                  id="path-1"
                                  points="22.407 43.4168 22.407 0.6878 0.5635 0.6878 0.5635 43.4168 22.407 43.4168"
                                />
                              </defs>
                              <g
                                fill="none"
                                fillRule="evenodd"
                                id="Modul-forslag"
                                stroke="none"
                                strokeWidth="1"
                              >
                                <g
                                  id="NAV-logo"
                                >
                                  <g
                                    id="Page-1-Copy"
                                  >
                                    <polygon
                                      fill="white"
                                      id="Fill-3"
                                      points="0 121.3588 17.265 78.6298 33.854 78.6298 16.611 121.3588"
                                    />
                                    <polygon
                                      fill="white"
                                      id="Fill-5"
                                      points="213.044 121.3588 230.088 78.6298 239.132 78.6298 222.089 121.3588"
                                    />
                                    <g
                                      id="Group-9"
                                      transform="translate(246.000000, 77.942000)"
                                    >
                                      <mask
                                        fill="white"
                                        id="mask-2"
                                      >
                                        <use
                                          xlinkHref="#path-1"
                                        />
                                      </mask>
                                      <g
                                        id="Clip-8"
                                      />
                                      <polygon
                                        fill="white"
                                        id="Fill-7"
                                        mask="url(#mask-2)"
                                        points="0.5635 43.4168 17.6045 0.6878 22.4075 0.6878 5.3645 43.4168"
                                      />
                                    </g>
                                    <path
                                      d="M125.3091,168.942 C78.6681,168.942 40.8491,131.125 40.8491,84.477 C40.8491,37.824 78.6681,0 125.3091,0 C171.9671,0 209.7901,37.824 209.7901,84.477 C209.7901,131.125 171.9671,168.942 125.3091,168.942 Z M197.3604,78.6298 L182.3444,78.6298 C182.3444,78.6298 181.3094,78.6298 180.9434,79.5438 L172.6334,104.9828 L164.3304,79.5438 C163.9644,78.6298 162.9234,78.6298 162.9234,78.6298 L134.0514,78.6298 C133.4264,78.6298 132.9024,79.1518 132.9024,79.7728 L132.9024,88.4118 C132.9024,81.5588 125.6104,78.6298 121.3404,78.6298 C111.7784,78.6298 105.3774,84.9278 103.3844,94.5028 C103.2764,88.1508 102.7484,85.8748 101.0374,83.5438 C100.2514,82.4018 99.1154,81.4418 97.8784,80.6478 C95.3314,79.1558 93.0444,78.6298 88.1294,78.6298 L82.3584,78.6298 C82.3584,78.6298 81.3154,78.6298 80.9474,79.5438 L75.6964,92.5568 L75.6964,79.7728 C75.6964,79.1518 75.1764,78.6298 74.5524,78.6298 L61.1984,78.6298 C61.1984,78.6298 60.1674,78.6298 59.7924,79.5438 L54.3334,93.0748 C54.3334,93.0748 53.7884,94.4278 55.0344,94.4278 L60.1674,94.4278 L60.1674,120.2118 C60.1674,120.8518 60.6714,121.3588 61.3144,121.3588 L74.5524,121.3588 C75.1764,121.3588 75.6964,120.8518 75.6964,120.2118 L75.6964,94.4278 L80.8564,94.4278 C83.8174,94.4278 84.4444,94.5088 85.5964,95.0458 C86.2904,95.3078 86.9154,95.8378 87.2564,96.4488 C87.9544,97.7628 88.1294,99.3408 88.1294,103.9938 L88.1294,120.2118 C88.1294,120.8518 88.6434,121.3588 89.2784,121.3588 L101.9664,121.3588 C101.9664,121.3588 103.4004,121.3588 103.9674,119.9428 L106.7794,112.9928 C110.5184,118.2298 116.6724,121.3588 124.3204,121.3588 L125.9914,121.3588 C125.9914,121.3588 127.4344,121.3588 128.0054,119.9428 L132.9024,107.8148 L132.9024,120.2118 C132.9024,120.8518 133.4264,121.3588 134.0514,121.3588 L147.0034,121.3588 C147.0034,121.3588 148.4324,121.3588 149.0064,119.9428 C149.0064,119.9428 154.1864,107.0818 154.2064,106.9848 L154.2144,106.9848 C154.4134,105.9148 153.0614,105.9148 153.0614,105.9148 L148.4384,105.9148 L148.4384,83.8468 L162.9834,119.9428 C163.5514,121.3588 164.9834,121.3588 164.9834,121.3588 L180.2844,121.3588 C180.2844,121.3588 181.7244,121.3588 182.2924,119.9428 L198.4174,80.0138 C198.9754,78.6298 197.3604,78.6298 197.3604,78.6298 Z M132.9024,105.9148 L124.2024,105.9148 C120.7394,105.9148 117.9224,103.1108 117.9224,99.6438 C117.9224,96.1828 120.7394,93.3608 124.2024,93.3608 L126.6354,93.3608 C130.0894,93.3608 132.9024,96.1828 132.9024,99.6438 L132.9024,105.9148 Z"
                                      fill="white"
                                      id="Combined-Shape"
                                    />
                                  </g>
                                </g>
                              </g>
                            </svg>
                          </NavLogoTransparent>
                        </a>
                        <styled.div>
                          <div
                            className="c6 c7"
                          />
                        </styled.div>
                        <styled.div>
                          <div
                            className="c8 c9"
                          >
                            <span>
                              ui:app-headerTitle
                            </span>
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                    <Styled(Lenke)
                      data-testid="c-header__highcontrast-link-id"
                      href="#highContrast"
                      onClick={[Function]}
                    >
                      <Lenke
                        className="c10 c11"
                        data-testid="c-header__highcontrast-link-id"
                        href="#highContrast"
                        onClick={[Function]}
                      >
                        <a
                          className="lenke c10 c11"
                          data-testid="c-header__highcontrast-link-id"
                          href="#highContrast"
                          onClick={[Function]}
                        >
                          ui:highContrast
                        </a>
                      </Lenke>
                    </Styled(Lenke)>
                    <styled.div>
                      <div
                        className="c12 c13"
                      >
                        <styled.div>
                          <div
                            className="c14 c15"
                          >
                            <FontAwesomeIcon
                              border={false}
                              className=""
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    448,
                                    512,
                                    Array [],
                                    "f007",
                                    "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
                                  ],
                                  "iconName": "user",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-user fa-w-14 "
                                data-icon="user"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 448 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                          </div>
                        </styled.div>
                        <styled.div>
                          <div
                            className="c16 c17"
                          />
                        </styled.div>
                        <styled.div>
                          <div
                            className="c6 c7"
                          />
                        </styled.div>
                        <styled.div>
                          <div
                            className="c18 c19"
                          >
                            <styled.span>
                              <span
                                className="c20 c21"
                              >
                                mockUsername
                              </span>
                            </styled.span>
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </header>
                </styled.header>
              </Header>
              <Alert
                message="mockErrorMessage"
                onClose={[Function]}
                status="ERROR"
                targetRef={
                  Object {
                    "current": null,
                  }
                }
              >
                <Styled(AlertStripe)
                  className="status-ERROR"
                  role="alert"
                  style={Object {}}
                  type="feil"
                >
                  <AlertStripe
                    className="c22 c23 status-ERROR"
                    form="default"
                    role="alert"
                    size="1.5em"
                    style={Object {}}
                    type="feil"
                  >
                    <div
                      className="alertstripe c22 c23 status-ERROR alertstripe--feil"
                      role="alert"
                      style={Object {}}
                    >
                      <span
                        className="alertstripe__ikon"
                      >
                        <span
                          className="sr-only"
                        >
                          feil
                        </span>
                        <Icon
                          aria-label="feil-ikon"
                          kind="feil-sirkel-fyll"
                          role="img"
                          size="1.5em"
                        >
                          <svg
                            aria-label="feil-ikon"
                            focusable="false"
                            height="1.5em"
                            kind="feil-sirkel-fyll"
                            role="img"
                            viewBox="0 0 24 24"
                            width="1.5em"
                          >
                            <g
                              fill="none"
                              fillRule="evenodd"
                            >
                              <path
                                d="M11.999 0C5.395 0 .013 5.372 0 11.976a11.923 11.923 0 0 0 3.498 8.493A11.925 11.925 0 0 0 11.977 24H12c6.603 0 11.986-5.373 12-11.978C24.013 5.406 18.64.012 11.999 0z"
                                fill="#A13A28"
                                fillRule="nonzero"
                              />
                              <path
                                d="M12 10.651l3.372-3.372a.954.954 0 1 1 1.349 1.35L13.349 12l3.372 3.372a.954.954 0 1 1-1.35 1.349L12 13.349 8.628 16.72a.954.954 0 1 1-1.349-1.35L10.651 12 7.28 8.628A.954.954 0 1 1 8.63 7.28L12 10.651z"
                                fill="#FFF"
                                fillRule="nonzero"
                              />
                            </g>
                          </svg>
                        </Icon>
                      </span>
                      <Normaltekst
                        className="alertstripe__tekst"
                        tag="div"
                      >
                        <TypografiBase
                          className="alertstripe__tekst"
                          tag="div"
                          type="normaltekst"
                        >
                          <div
                            className="typo-normal alertstripe__tekst"
                          >
                            mockErrorMessage
                            <Styled(FilledVersionRemoveCircle)
                              data-testid="c-alert__close-icon"
                              onClick={[Function]}
                            >
                              <FilledVersionRemoveCircle
                                className="c24 c25"
                                data-testid="c-alert__close-icon"
                                onClick={[Function]}
                              >
                                <svg
                                  className="c24 c25"
                                  contentScriptType="text/ecmascript"
                                  contentStyleType="text/css"
                                  data-testid="c-alert__close-icon"
                                  enableBackground="new 0 0 24 24"
                                  height={24}
                                  id="Filled_Version"
                                  onClick={[Function]}
                                  preserveAspectRatio="xMidYMid meet"
                                  version="1.1"
                                  viewBox="0 0 24 24"
                                  width={24}
                                  x="0px"
                                  xmlns="http://www.w3.org/2000/svg"
                                  y="0px"
                                  zoomAndPan="magnify"
                                >
                                  <path
                                    d="M11.499,0C5.17,0,0.012,5.148,0,11.477c-0.006,3.072,1.184,5.962,3.352,8.139c2.168,2.176,5.054,3.378,8.126,3.384H11.5  c6.328,0,11.487-5.149,11.5-11.479C23.012,5.181,17.863,0.012,11.499,0z M11.5,22.5L11.5,22.5L11.5,22.5L11.5,22.5z M16.088,15.398  c0.194,0.195,0.193,0.512-0.002,0.707c-0.098,0.098-0.225,0.146-0.353,0.146c-0.127,0-0.256-0.049-0.354-0.146l-3.882-3.898  l-3.896,3.882c-0.098,0.098-0.226,0.146-0.354,0.146c-0.128,0-0.255-0.049-0.353-0.146c-0.195-0.196-0.195-0.513,0.001-0.707  l3.897-3.882L6.91,7.602C6.715,7.406,6.716,7.09,6.912,6.896C7.107,6.699,7.424,6.7,7.619,6.897l3.881,3.897l3.896-3.881  c0.195-0.195,0.513-0.195,0.707,0.001c0.195,0.195,0.195,0.512-0.001,0.707l-3.896,3.882L16.088,15.398z"
                                    fill="#3e3832"
                                  />
                                </svg>
                              </FilledVersionRemoveCircle>
                            </Styled(FilledVersionRemoveCircle)>
                          </div>
                        </TypografiBase>
                      </Normaltekst>
                    </div>
                  </AlertStripe>
                </Styled(AlertStripe)>
              </Alert>
              <SessionMonitor
                targetRef={
                  Object {
                    "current": null,
                  }
                }
              >
                <div
                  data-testid="c-sessionMonitor"
                  role="application"
                />
              </SessionMonitor>
              <styled.main
                id="main"
                role="main"
                targetRef={
                  Object {
                    "current": null,
                  }
                }
              >
                <main
                  className="c26 c27"
                  id="main"
                  role="main"
                >
                  <div
                    id="TEST_CHILD"
                  />
                </main>
              </styled.main>
              <Footer
                footerOpen={false}
                params={Object {}}
                targetRef={
                  Object {
                    "current": null,
                  }
                }
              >
                <styled.footer
                  className=""
                  role="contentinfo"
                >
                  <footer
                    className="c28 c29"
                    role="contentinfo"
                  >
                    <styled.div
                      className="fullWidth"
                    >
                      <div
                        className="c30 c31 fullWidth"
                      >
                        <div
                          className="footerButtonClosed"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="button"
                          tabIndex={0}
                        />
                      </div>
                    </styled.div>
                  </footer>
                </styled.footer>
              </Footer>
            </ResizeDetector>
          </ErrorBoundary>
        </div>
      </styled.div>
    </Component>
  </he>
</TopContainer>
`;
